libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1142,17 +1144,26 @@ class MapboxRouteLineApi(
         featureDataProvider: () -> List<RouteFeatureData>,
         alternativeRoutesMetadata: List<AlternativeRouteMetadata>
     ): Expected<RouteLineError, RouteSetValue> {
-        ifNonNull(newRoutes.firstOrNull()) { primaryRouteCandidate ->
+        val distinctNewRoutes = newRoutes.distinctBy { it.id }
```

Comment:
Could it be problematic that here the ID is being used to distinguish routes but below on line 1157 `isSame` route is being used?

Reply:
The latter is more precise, which is a good thing here. We could possibly relax that and just use the ID, but I don't think it's worth refactoring just that now. At some point we could look at the general usage of `DirectionsRoute.isSameRoute` given that now we have the `NavigationRoutes#id`.